proto/viam/app/data/v1/data.proto

```diff
@@ -80,6 +80,18 @@ message Filter {
   repeated string org_ids = 11;
   repeated string mime_type = 12;
   CaptureInterval interval = 13;
+  TagsFilterType tags_filter_type = 14;
+}
+
+// TagsFilterType specifies how data can be filtered based on tags
+enum TagsFilterType {
+  TAGS_FILTER_TYPE_UNSPECIFIED = 0;
+  // TAGS_FILTER_TYPE_MATCH_BY_OR specifies documents matched based on OR on inputted string tags should be returned
```

Comment:
nit to make clear this is paired together with `tags` above: 'specifies documents matched (using logical OR) on the Filter `tags` field'